language_model/ngram_language_model.py

```diff
@@ -107,13 +96,24 @@ def event_handler(event):
 
     trainer = paddle.trainer.SGD(cost, parameters, adadelta_optimizer)
     trainer.train(
-        paddle.batch(ptb.ngram_train(word_dict, 6), batch_size),
+        paddle.batch(paddle.dataset.imikolov.train(word_dict, 6), batch_size),
         num_passes=30,
         event_handler=event_handler)
 
 
-def predict(word_dict, voc_dim):
-    prediction_layer = ngrm_network(voc_dim)
+def generate_sequence(word_dict, voc_dim):
+    """
+    the generating process as follows:
+      1 construct the ngram netword
+      2 read the model which is trained before
+      3 read the test data by ins_iter which is a five-element couple
+      4 generate the next word according to the last four words 
+          until the word count is equal to max_word_num or 
+          the next word is <e>
+      5 so far, we have completed one five-element couple predicting,
+        and then continue to iterate the next couple
+    """
```

Comment:
The explanation is expected to explain the idea of the one-way search algorithm.